### PR TITLE
Add -F <pidfile> flag to write manage a pidfile

### PR DESCRIPTION
**Pull request by burke**
Does what it says on the tin basically. I wanted to use `-p` or `-P`, but both were taken, so I settled for `-F`.

### CODE REVIEW

**Comment by AntonioMeireles** [comment]
@burke  :+1:

File: src/xhyve.c [hunk 1]
```diff
@@ -773,6 +776,57 @@ firmware_parse(const char *opt) {
...
+	char pid_str[21];
+
+	if (pidfile == NULL)
+		return 0;
+
+	pid = getpid();
+
+	error = sprintf(pid_str, "%d", pid);
+	if (error < 0)
+		return -1;
```
Comment (jeremyhu) [inline src/xhyve.c]: I don't think that all these arbitrary (and possibly overlapping with elsewhere) exit codes are a good idea.  It's not clear what each ones value means.  I suggest writing out the error to stderr and using EXIT_FAILURE or possibly using sysexits.h if you want more verbosity.

Comment (burke) [inline src/xhyve.c]: Agreed; updated the PR.

**Comment by jeremyhu** [comment]
Thanks.  This has been merged into https://github.com/xhyve-xyz/xhyve

**Comment by burke** [comment]
I'm not sure which enum you're talking about – maybe the one for vmexit handlers? I couldn't find precedent for that in these setup handlers, where the return value is just used to determine whether to `exit(1)` from `main`.

Anyway, I revised the PR to return `errno` instead of `-1`. If I've still missed your point, could you please point me toward the enum you're referencing?

File: config.mk [hunk 2]
```diff
@@ -34,6 +34,7 @@ CFLAGS_OPT := \
 CFLAGS_WARN := \
   -Weverything \
   -Werror \
+	-Wno-deprecated-declarations \
```
Comment (rickard-von-essen) [inline config.mk]: I'll like if we started to align spaces vs. tabs.

Comment (jeremyhu) [inline config.mk]: Also, don't hide them.  Just demote them to warnings.  They're there for a reason.  Use -Wno-error=deprecated-declarations

Comment (burke) [inline config.mk]: Oh hah, I didn't realize this PR was from my master branch.

**Pull request closed without merging**